src/command/client/search.rs

```diff
@@ -305,14 +305,14 @@ fn remove_char_from_input(app: &mut State, i: usize) -> char {
 
 #[allow(clippy::too_many_lines)]
 async fn key_handler(
-    input: Key,
+    input: TermEvent,
     search_mode: SearchMode,
     db: &mut impl Database,
     app: &mut State,
 ) -> Option<String> {
     match input {
-        Key::Esc | Key::Ctrl('c' | 'd' | 'g') => return Some(String::from("")),
-        Key::Char('\n') => {
+        TermEvent::Key(Key::Esc | Key::Ctrl('c' | 'd' | 'g')) => return Some(String::from("")),
```

Comment:
kind of horrible the wrapping all the keys in TermEvents

Another option would be first split into Key/Mouse events and handle separately, but that would lead to duplication of the up/down handler